core/src/main/java/de/danoeh/antennapod/core/service/playback/ExoPlayerWrapper.java

```diff
@@ -312,8 +317,8 @@ public int getSelectedAudioTrack() {
             if (track == null) {
                 continue;
             }
-            if (availableFormats.contains(track.getSelectedFormat())) {
-                return availableFormats.indexOf(track.getSelectedFormat());
+            if (availableFormats.contains(track.getFormat(0))) {
```

Comment:
Why are you not casting the track variable to `ExoTrackSelection` and call the `getSelectedFormat()` method from there? The `AdaptiveTrackSelection` and `RandomTrackSelection` implementations are not always using the first format in the `Format` array.

Reply:
I have no idea why I did that lol, I'll fix it.

Reply:
Hmm, simply down-casting without a check sounds dangerous. Even though their current implementation always returns a `ExoTrackSelection`, it might change in the future. They probably made two different interfaces for a reason.

Reply:
Interesting. What would you suggest be done, then? Should this change be reverted or kept? Or should a check be added?

Reply:
Have you seen the [How to get selected Format with TracksInfo? #9674](https://github.com/google/ExoPlayer/issues/9674) question in the ExoPlayer repo? tonihei has written in one of the comment:

> (Exo)TrackSelection.getSelectedFormat() was never meant to be called by app code as it's not thread safe and also not aligned with the rest of the player state.
>
> The simplest way to listen to adaptive video format changes is to listen to Player.Listener.onVideoSizeChanged, or alternatively to AnalyticsListener.onDownstreamFormatChanged (which contains more details like the actual Format and the reason for the change). You can also obtain the current video size by calling Player.getVideoSize().

So this sounds like a bigger change (maybe something for another pull request) if it should be done in a proper way?